pkg/asset/installconfig/alibabacloud/metadata.go

```diff
@@ -0,0 +1,27 @@
+package alibabacloud
+
+// Metadata holds additional metadata for InstallConfig resources that
+// does not need to be user-supplied (e.g. because it can be retrieved
+// from external APIs).
+type Metadata struct {
+	client     *Client
```

Comment:
This stored `*Client` is not being utilized, instead `NewClient` is called throughout the code.